include/vrv/doc.h

```diff
@@ -409,6 +423,11 @@ class Doc : public Object {
     /** Record notation type for document */
     data_NOTATIONTYPE m_notationType;
 
+    /** The expansion map indicates which xmlId has been repeated (expanded) elsewhere */
+    std::vector<std::vector<std::string> > m_expansionMap;
```

Comment:
I think we need to add a new ExpansionMap class to handle this. The we can move some of the methods in Doc there, which would make it clearer.

Also, I can imaging you could use a <std::map> to map each ids with the expanded ones, but maybe I am overlooking something.

Reply:
Ok, I can create a new file called ExpansionMap (or inside doc.cpp?).

Re: using std::map for this: as the lookup is not always pairwise (when there are more than one repetition and we have often different numbers of repetitions for the elements). 

Do you think of this structure std::map<std::string, std::vector<std::string>> that has each element as key and a complete list of element Ids as value: 
"orig1"; <"orig1","rend2">
"rend2"; <"orig1","rend2">
When adding a new pair "orig1","rend3" would result in this:
"orig1"; <"orig1","rend2","rend3">
"rend2"; <"orig1","rend2","rend3">
"rend3"; <"orig1","rend2","rend3">

That is slower in construction, but faster in retrieving the content. Is that, what you have in mind?

Reply:
> I can create a new file called ExpansionMap

Yes, create a new file.

> Do you think of this structure std::map<std::string, std::vectorstd::string> 

Yes, exactly.

Reply:
Just one more question: do you see the `std::map<std::string, std::vectorstd::string>` to live in that `expansionMap` class, or could that stay within `doc` (which seems simpler to me)? If not, how would you link the expansionMap to the doc?

Reply:
In ExpansionMap. Doc would simply have one ExpansionMap member. It does not make any difference, it is mostly not to overload the Doc class, which is already quite dense. (In the end, it is mostly to have the methods in the ExpansionMap and not in Doc)